frontends/common/constantFolding.cpp

```diff
@@ -23,50 +23,49 @@ namespace P4 {
 
 const IR::Expression* DoConstantFolding::getConstant(const IR::Expression* expr) const {
     CHECK_NULL(expr);
-    auto cst = get(constants, expr);
-    if (cst != nullptr)
-        return cst;
-    if (expr->is<IR::Constant>())
+    if (expr->is<IR::Constant>()) {
         return expr;
-    if (expr->is<IR::BoolLiteral>())
+    }
+    if (expr->is<IR::BoolLiteral>()) {
```

Comment:
Seems like this could be just `expr->is<IR::Literal>()` to get Constant, BoolLiteral and StringLiteral